python/rascaline-torch/rascaline/torch/utils/power_spectrum.py

```diff
@@ -0,0 +1,105 @@
+
+
+# from ._classes import (Labels, TensorBlock, TensorMap, CalculatorBase, IntoSystem)
+
+# For details what is happening here take a look an `rascaline.torch.calculators`.
+
+# Step 1: create te `_classes` module as an empty module
+spec = importlib.util.spec_from_loader(
+    "rascaline.torch.utils.power_spectrum._classes",
+    loader=None,
+)
+module = importlib.util.module_from_spec(spec)
+# This module only exposes a handful of things, defined here. Any changes here MUST also
+# be made to the `equistore/operations/_classes.py` file, which is used in non
+# TorchScript mode.
+module.__dict__["Labels"] = Labels
+module.__dict__["TensorBlock"] = TensorBlock
+module.__dict__["TensorMap"] = TensorMap
+module.__dict__["CalculatorBase"] = CalculatorBase
+module.__dict__["IntoSystem"] = IntoSystem
+
+# register the module in sys.modules, so future import find it directly
+sys.modules[spec.name] = module
+
+# # TODO: explain
+# spec = importlib.util.spec_from_file_location(
+#     "rascaline.torch.utils._dispatch",
+#     rascaline.utils._dispatch.__file__,
+# )
+# module = importlib.util.module_from_spec(spec)
+# sys.modules[spec.name] = module
+# spec.loader.exec_module(module)
+
+
+# Step 2: create a module named `rascaline.torch.utils.power_spectrum` using code from
+# `rascaline.utils.power_spectrum`
+spec = importlib.util.spec_from_file_location(
+    "rascaline.torch.utils.power_spectrum",
+    rascaline.utils.power_spectrum.__file__,
+)
+
+module = importlib.util.module_from_spec(spec)
+sys.modules[spec.name] = module
+spec.loader.exec_module(module)
+
+
+# Define the class to avoid lookup and recursion problems
+PowerSpectrumBase = module.PowerSpectrum
+
+
+class PowerSpectrum(torch.nn.Module, PowerSpectrumBase):
+    """
+    Torch version of the general power spectrum of one or of two calculators.
+
+    The class provides :py:meth:`PowerSpectrum.forward` and the integration with
+    :py:class:`torch.nn.Module`. For more details see
+    :py:class:`rascaline.utils.PowerSpectrum`.
+
+    :param calculator_1: first calculator
+    :param calculator_1: second calculator
+    :raises ValueError: If other calculators than
+        :py:class:`rascaline.SphericalExpansion` or
+        :py:class:`rascaline.LodeSphericalExpansion` are used.
+    :raises ValueError: If ``'max_angular'`` of both calculators is different
+    """
+
+    def __init__(
+        self,
+        calculator_1: CalculatorBase,
+        calculator_2: Optional[CalculatorBase] = None,
+    ):
+        torch.nn.Module.__init__(self)
+        PowerSpectrumBase.__init__(
+            self,
+            calculator_1=calculator_1,
+            calculator_2=calculator_2,
+        )
+
+    def forward(
+        self,
+        systems: Union[IntoSystem, List[IntoSystem]],
+        gradients: Optional[List[str]] = None,
+        use_native_system: bool = True,
+    ) -> TensorMap:
+        """forward just calls :py:meth:`PowerSpectrum.compute`"""
+
+        return self.compute(
```

Comment:
should we do something like this to avoid issues with a possible `compute` method on torch.nn.Module?

```suggestion
        return PowerSpectrumBase.compute(self,
```

Reply:
Seems like torch doesn't like this.